File: backend/src/main/java/com/example/demo/model/ApartmentDevice.java
package com.example.demo.model;

public class ApartmentDevice extends Device {

        private Apartment apartment;
        public ApartmentDevice(int id, String name, int consumesEnergy, EnergyCurve energyCurve, Apartment apartment, float windSensitivity, float lightSensitivity, float temperatureSensitivity, float precipitationSensitivity) {
            super(id, name, consumesEnergy, energyCurve, windSensitivity, lightSensitivity, temperatureSensitivity, precipitationSensitivity);
            this.apartment = apartment;
        }

        public Apartment getApartment() {
            return apartment;
        }

        public void setApartment(Apartment apartment) {
            this.apartment = apartment;
        }

        public String toString() {
            return "ApartmentDevice{" +
                    super.toString()+
                    "apartment=" + apartment +
                    '}';
        }
}

File: frontend/src/app/models/apartment_device.ts
import { Apartment } from './apartment';
import { EnergyCurve } from './energy_curve';

export interface ApartmentDevice
{
    id: number;
    name: string;
    consumesEnergy: number;
    energy_curve: EnergyCurve;
    apartment: Apartment | null;
    lightSensitivity: number;
    windSensitivity: number;
    precipitationSensitivity: number;
    temperatureSensitivity: number;
}